core/src/main/java/org/elasticsearch/search/highlight/HighlightBuilder.java

```diff
@@ -289,7 +346,95 @@ public static HighlightBuilder fromXContent(QueryParseContext parseContext) thro
         return highlightBuilder;
     }
 
+    public void parse(XContentParser parser, SearchContext context) throws Exception {
```

Comment:
Yes please do make build() public. SearchParseElement will be deleted at the end of the refactoring so there should be no new implementations as part of the refactoring